pkg/auth/certs/provider.go

```diff
@@ -15,10 +17,16 @@ type CertificateProvider interface {
 
 	// GetCertificateForIdentity gives the certificate and intermediates required
 	// to send as trust chain for a certain identity as well as a private key
-	GetCertificateForIdentity(identity string) (*tls.Certificate, error)
+	GetCertificateForNumericIdentity(id identity.NumericIdentity) (*tls.Certificate, error)
```

Comment:
personally, i would name this method (and maybe even the conversion methods) just `GetCertificateForIdentity` - without the `Numeric` part